Unix/tools/auth_tests_setup.sh

```diff
@@ -37,34 +40,60 @@ do
 done
 
 #
-# if required env variables don't exist, then quit
+# Setup the NTLM auth file for client and server. They must be different files because 
+# if the differeing permissions, as the server file must be owned by root and the client by
+# the user.
+#
+# If there is already a cred file present we defer to it. In configure we possibly copied over
+# a prebuilt cred file for the server from Unix/etc/creds. If such a file exists, it must contain the test
+# credentials.
+#
+# If the server cred file isn't present we build one with the test credentials, derived from OMI_USER and 
+# OMI_PASSWORD env variables.
 #
+# We create the cred file for the client and point to it with the NTLM_USER_FILE env variable.  This prevents us 
+# from clobbering the credentials in ~/.omi/
+#
+# Once we have created the cred files, they are static unless removed, edited, or configured via
+# configure. We always set the ownership and r/w permissions.
+
 if [ "x${username}" != "x" -a "x${userpasswd}" != "x" ]; then
     SUDO_PATH=`which sudo`
     export SUDO_PATH
 
 ## NTLM setup
-
     
-    ntlm_cred_dir=$OUTPUTDIR/tmp/creds
-    ntlm_file=$OUTPUTDIR/tmp/creds/ntlm
+    # client side creds
+    echo "build cred files"
+
+    client_ntlm_cred_dir=$outputdir/tmp/creds
+    client_ntlm_file=$outputdir/tmp/creds/ntlm
     hostname=`uname -n`
-    if [ -f $ntlm_cred_dir ]; then
-        sudo rm -rf $ntlm_cred_dir
-    fi
+    if [ ! -f $client_ntlm_file ]; then
+       echo "build client auth file " $client_ntlm_file
 
-    mkdir $ntlm_cred_dir
-    echo $hostname":"$username":"$userpasswd > $ntlm_file
+       mkdir -p $outputdir/etc
+       mkdir -p $client_ntlm_cred_dir
+       echo $hostname":"$username":"$userpasswd > $client_ntlm_file
 
-    chmod 700 $ntlm_cred_dir
-    chmod 600 $ntlm_file
+    fi
+    chmod 700 $client_ntlm_cred_dir
+    chmod 600 $client_ntlm_file
 
-    NTLM_USER_FILE=$ntlm_file
+    NTLM_USER_FILE=$client_ntlm_file
     export NTLM_USER_FILE
-    NTLM_DOMAIN=$hostname
-    export NTLM_DOMAIN
+
+    # server side 
+    # configure already created the dir
+
+    server_ntlm_file=$outputdir/etc/creds/ntlm
+    if sudo test ! -f $server_ntlm_file ; then
+       echo "build server auth file " $server_ntlm_file
+       sudo cp $client_ntlm_file $server_ntlm_file
+    else 
+       echo "do not build server auth file " $server_ntlm_file
+    fi
+    sudo chown root:root $server_ntlm_file
```

Comment:
Where is the "root:root" requirement coming in? If code is validating that, it should be mentioned in my spec. Please give me a filename/filenames and line numbers for this, thanks.